nutanix-cloud-native/cluster-api-provider-nutanix
# Feat/secret mgmt

**What this PR does / why we need it**:
- Change the way credentials are handled:
     -  Allow credentials to be defined on nutanixCluster and Manager
     - Add Prism Central attribute on nutanixCluster
     - Remove endpoint, port flags from secret
     - Implement optional insecure mode
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
N/A
**How Has This Been Tested?**:
- Create CAPI cluster with new cluster-template.yml file
- Use different user/pass at manager and cluster level and check VM creation
- Use insecure false on PC with invalid certs and check failure

**Special notes for your reviewer**:

**Release note**:
```release-note
- Change the way credentials are handled:
     - Allow credentials to be defined on nutanixCluster and Manager
     - Add Prism Central attribute on nutanixCluster
     - Remove endpoint, port flags from secret
     - Implement optional insecure mode
```
